### PR TITLE
[Az.CosmosDB] Add table and gremlin APIs for point-in-time restore

### PR DESCRIPTION
**Pull request by kshittiz7**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/CosmosDB/CosmosDB/ChangeLog.md [hunk 1]
```diff
@@ -37,6 +37,13 @@
 ## Version 1.7.0
 * Introduced support for client encryption key resource management required for CosmosDB Client-Side Encryption by adding support for creating, updating and retrieving client encryption keys with following cmdlets: `Get-AzCosmosDbClientEncryptionKey`, `New-AzCosmosDbClientEncryptionKey` and `Update-AzCosmosDbClientEncryptionKey`
 
+## Version 1.6.0
```
Comment (VeryEarly) [inline src/CosmosDB/CosmosDB/ChangeLog.md:40]: please move release notes under line 21

File: src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBGremlinDatabaseToRestore.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBGremlinDatabaseToRestore"), OutputType(typeof(PSGremlinDatabaseToRestore))]
+    public class NewAzCosmosDBGremlinDatabaseToRestore : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = Constants.RestoreGremlinDatabaseNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.RestoreGraphNamesHelpMessage)]
+        public string[] GraphNames { get; set; }
```
Comment (VeryEarly) [inline src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBGremlinDatabaseToRestore.cs:29]:
```suggestion
        public string[] GraphName { get; set; }
```

File: src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBTableToRestore.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBTableToRestore"), OutputType(typeof(PSTablesToRestore))]
+    public class NewAzCosmosDBTableToRestore : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = false, HelpMessage = Constants.RestoreTableNamesHelpMessage)]
+        public string[] TableNames { get; set; }
```
Comment (VeryEarly) [inline src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBTableToRestore.cs:25]:
```suggestion
        public string[] TableName { get; set; }
```

File: src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBGremlinRestorableGraph.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.AccountInstanceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseAccountInstanceId { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.DatabaseResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseRId { get; set; }
```
Comment (VeryEarly) [inline src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBGremlinRestorableGraph.cs:37]:
```suggestion
        public string DatabaseResourceId { get; set; }
```

File: src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBGremlinRestorableGraph.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSet, HelpMessage = Constants.RestorableGremlinGraphsFeedEndTimeHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string EndTime { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSet, StringComparison.Ordinal))
+            {
+                // id is in the format: /subscriptions/<subscriptionId>/providers/Microsoft.DocumentDB/locations/<location>/restorableDatabaseAccounts/<DatabaseAccountInstanceId>/restorableGremlinDatabases/<Id>
```
Comment (VeryEarly) [inline src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBGremlinRestorableGraph.cs:55]: NIT: you can take advantage of this class:
https://github.com/Azure/azure-powershell-common/blob/e18a2440ce45fde879416611063fea9b0b7ed84d/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs

**Review by VeryEarly — CHANGES_REQUESTED** [review]

File: src/CosmosDB/CosmosDB/help/Get-AzCosmosDBGremlinGraphBackupInformation.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.CosmosDB.dll-Help.xml
+Module Name: Az.CosmosDB
+online version:
```
Comment (VeryEarly) [inline src/CosmosDB/CosmosDB/help/Get-AzCosmosDBGremlinGraphBackupInformation.md:4]: please add online version for all help markdown

**Review by VeryEarly — CHANGES_REQUESTED** [review]
please record new test cases

File: src/CosmosDB/CosmosDB/ChangeLog.md (hunk 1 above)
Comment (kshittiz7) [reply src/CosmosDB/CosmosDB/ChangeLog.md:40]: should I bump version from latest 1.9.0 to 1.9.1?

File: src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBGremlinDatabaseToRestore.cs (hunk 2 above)
Comment (kshittiz7) [reply src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBGremlinDatabaseToRestore.cs:29]: We already have this change in preview version and we are just making it stable. Also, we are following similar convention for existing APIs of sql and mongo so we decided to use same for these new ones as well to remain consistent.

File: src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBTableToRestore.cs (hunk 3 above)
Comment (kshittiz7) [reply src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBTableToRestore.cs:25]: We already have this change in preview version and we are just making it stable. Also, we are following similar convention for existing APIs of sql and mongo so we decided to use same for these new ones as well to remain consistent.

File: src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBGremlinRestorableGraph.cs (hunk 4 above)
Comment (kshittiz7) [reply src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBGremlinRestorableGraph.cs:37]: We already have this change in preview version and we are just making it stable. Also, we are following similar convention for existing APIs of sql and mongo so we decided to use same for these new ones as well to remain consistent.

File: src/CosmosDB/CosmosDB/ChangeLog.md (hunk 1 above)
Comment (VeryEarly) [reply src/CosmosDB/CosmosDB/ChangeLog.md]: we will calculate the correct version during release, no need to specify one in code changes

File: src/CosmosDB/CosmosDB/help/Get-AzCosmosDBGremlinGraphBackupInformation.md (hunk 6 above)
Comment (kshittiz7) [reply src/CosmosDB/CosmosDB/help/Get-AzCosmosDBGremlinGraphBackupInformation.md:4]: where can I get the online version for all help files?

File: src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBTableToRestore.cs (hunk 3 above)
Comment (kshittiz7) [reply src/CosmosDB/CosmosDB/CosmosDBAccount/NewAzCosmosDBTableToRestore.cs]: Made the change

**Comment by pjohari-ms** [comment]
> please record new test cases

@VeryEarly Is there an SoP on how to re record existing test cases? If not, could you provide us a way to do this?

**Comment by kshittiz7** [comment]
> please record new test cases

I ran the test locally on my machine, after setting up environment variables as described in guideline. I am able to run all test successfully on my machine and generate session records, but on pipeline it says 3 tests failed. can you help me with this? https://dev.azure.com/azclitools/public/_build/results?buildId=41022&view=logs&j=5264e576-3c6f-51f6-f055-fab409685f20&t=11b6e5a1-913e-50ac-35c6-54672a9d99ea

Also, it's using old Api version 08-15preview not latest 11-15 stable. Is there a way to update it?

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**